tests/sdk/utils.ts

```diff
@@ -19,7 +19,18 @@ export class TestCache implements FlagsmithCache {
     }
 }
 
-export const fetch = vi.fn(global.fetch)
+export const fetch = vi.fn((_, options) => {
+    const headers = options?.headers as Record<string, string>;
+    if (!headers) throw new Error('missing request headers')
+    const env = headers['X-Environment-Key'];
+    if (!env) return Promise.resolve(new Response("missing x-environment-key header", { status: 404 }))
+    if (env.startsWith('ser.')) {
+        return Promise.resolve(new Response(environmentJSON, { status: 200 }))
+    }
+    return Promise.resolve(new Response(identitiesJSON, { status: 200 }))
+});
```

Comment:
This is a lot of code now for a test util, and seems quite specific to the relevant tests. For example, it looks like we default to returning `identitiesJSON` if the key is a client side key. I have 2 issues with this:

 1. We shouldn't ever really be testing against the client side key. Perhaps with Node.js there are use cases (in SSR apps for example) to use a client side key, but I'm not sure. My general stance is that there are no genuine use cases for using a client side key in a server side SDK. 
 2. What about the flags endpoint?

Reply:
This is now changed to simulate a crude API: https://github.com/Flagsmith/flagsmith-nodejs-client/pull/178/files#diff-afedd75c48cb04d2e04d97213b7825890b2c721108999106a5936cba2993596dR22